web/tests/composebox_typeahead.test.js

```diff
@@ -500,6 +513,27 @@ function test(label, f) {
     });
 }
 
+run_test("stream_topic_link_syntax_test", () => {
+    assert.equal(ct.stream_topic_link_syntax("#**stream>", "topic"), "#**stream>topic** ");
+    assert.equal(ct.stream_topic_link_syntax("#**stream>to", "topic"), "#**stream>topic** ");
+    assert.equal(
+        ct.stream_topic_link_syntax("#**stream>t", "test `test` test"),
+        "[#stream>test `test` test](#narrow/stream/-1-unknown/topic/test.20.60test.60.20test) ",
+    );
+    assert.equal(
+        ct.stream_topic_link_syntax("#**stream>t", "test `test` test`s"),
+        "[#stream>test `test` test`s](#narrow/stream/-1-unknown/topic/test.20.60test.60.20test.60s) ",
+    );
+    assert.equal(
+        ct.stream_topic_link_syntax("#**stream>typeah", "error due to *"),
+        "[#stream>error due to *](#narrow/stream/-1-unknown/topic/error.20due.20to.20*) ",
+    );
+    assert.equal(
+        ct.stream_topic_link_syntax("#**stream>t", "*asterisk"),
+        "[#stream>*asterisk](#narrow/stream/-1-unknown/topic/*asterisk) ",
```

Comment:
When we aren't generating markdown link syntax ourselves, there is no need for a stream id, as we take the stream name from what the user had already typed. 
So they didn't require any id.

Reply:
stream ids are only required when we generate the markdown link syntax (since we need to generate the url ourselves).
That is why the first two didn't need them.